core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectLifecycle.scala

```diff
@@ -99,4 +102,36 @@ class BpmBillingProjectCreator(billingRepository: BillingRepository,
     } catch {
       case exception: Exception => Future.failed(exception)
     }
+
+  override def preDeletionSteps(projectName: RawlsBillingProjectName, ctx: RawlsRequestContext): Future[Unit] =
+    for {
+      _ <- billingRepository.getCreationStatus(projectName).flatMap {
+        case CreationStatuses.CreatingLandingZone =>
+          Future.failed(
+            new BillingProjectDeletionException(
+              RawlsErrorReport(
+                s"Billing project ${projectName.value} cannot be deleted because its landing zone is still being created"
+              )
+            )
+          )
+        case _ => Future.successful()
+      }
+      _ <- billingRepository.getLandingZoneId(projectName).flatMap {
+        case Some(landingZoneId) =>
+          val landingZoneResponse = blocking {
+            workspaceManagerDAO.deleteLandingZone(UUID.fromString(landingZoneId), ctx)
+          }
+          val errorReport = Option(landingZoneResponse.getErrorReport)
+          errorReport match {
```

Comment:
Nit: it would probably read a little cleaner to match directly on the `Option(landingZoneResponse.getErrorReport)`, instead of assigning it to a separate value, especially since you're immediately deconstructing it.
```scala
Option(landingZoneResponse.getErrorReport) match {
    case Some(errorReport) => ....
}
```